core/json/src/main/scala/net/liftweb/json/JsonParser.scala

```diff
@@ -315,11 +358,19 @@ object JsonParser {
             fieldNameMode = true
             blocks.poll
             return CloseArr
-          case c if Character.isDigit(c) || c == '-' =>
+          case c @ ('0' | '1' | '2' | '3' | '4' | '5' | '6' | '7' | '8' | '9' | '-') =>
```

Comment:
ORLY. Huh. I guess I haven't ever thought about this having such a significant performance impact, but yeah, makes sense.